### PR TITLE
combining repeating fucntion of UT together for workload

### LINKED ISSUE #1052: Abstract unified helper function to build fake service and fake workload
**Task description**:

In workload processing related UT, we need to create fake workloads and fake services in many places. Now different modules have their own creating logic, like building local hepler functions:

https://github.com/kmesh-net/kmesh/blob/main/pkg/controller/workload/workload_processor_test.go#L473

https://github.com/kmesh-net/kmesh/blob/main/pkg/controller/workload/workload_processor_test.go#L427

https://github.com/kmesh-net/kmesh/blob/main/pkg/controller/workload/workload_processor_test.go#L427

https://github.com/kmesh-net/kmesh/blob/main/pkg/controller/workload/cache/waypoint_cache_test.go#L127

or create them directly in the test function:

https://github.com/kmesh-net/kmesh/blob/main/pkg/controller/workload/cache/workload_cache_test.go#L51

This will result in a lot of cod redundancy and increate the difficulty of maintenance.

**Solution**:

Abstract a util lib which contains common tool functions such as `createFakeWorkload` and `createFakeService`, which can be called where needed, so that most scenarios can be convered.

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by Insomniac2904**
**What type of PR is this?**
cleanup

**What this PR does / why we need it**:
 single functions for creating fake workloads and services.

**Which issue(s) this PR fixes**:
Fixes #1052 

**Special notes for your reviewer**:
I have used `optional ...interface{}` to handle different ways of function calls.
I will remove the commented code if the changes gets approved.

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

### CODE REVIEW

File: pkg/controller/workload/cache/waypoint_cache_test.go [hunk 1]
```diff
@@ -112,59 +109,59 @@ func TestBasic(t *testing.T) {
...
-					},
-				},
-				HboneMtlsPort: 15008,
-			}
-		}
-	}
-
-	return w
-}
+// func createFakeWorkload(ip string, waypoint string) *workloadapi.Workload {
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/cache/waypoint_cache_test.go:112]: Delete this

File: pkg/controller/workload/common/fake_helpers.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
+package common
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/common/fake_helpers.go:1]: Add copyright

**Review by LiZhenCheng9527 — COMMENTED** [review]
Almost lgtm.
This PR  can be merged after updating.

File: pkg/controller/workload/common/fake_helpers.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+	workload := &workloadapi.Workload{
+		Uid:       "cluster0/" + rand.String(6),
+		Namespace: "ns",
+		Name:      "name",
+		Addresses: [][]byte{netip.MustParseAddr(ip).AsSlice()},
+		Waypoint:  w,
+	}
+
+	if len(optional) > 0 {
```
Comment (YaoZengzeng) [inline pkg/controller/workload/common/fake_helpers.go:86]: @Insomniac2904 Thanks for your contribution and greate code refator :)

There is only one suggestion. The assertion of options is a bit hardcoded. You can refer to the machanism for adding optional options in grpc-go and passs in option closure as needed, ref: https://github.com/grpc/grpc-go/blob/master/clientconn.go#L131

File: pkg/controller/workload/common/fake_helpers.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+			generatedUid = "cluster0//Pod/default/" + *name
+		}
+		uid = stringPtr(generatedUid)
+	}
+
+	// Create the workload object with default values
+	workload := &workloadapi.Workload{
+		Uid:               *uid,
+		Name:              *name,
+		Node:              *nodeName,
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/common/fake_helpers.go:149]: if params is string, nodeName is nil which raises the null pointer panic

File: pkg/controller/workload/common/fake_helpers.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+					},
+				},
+				HboneMtlsPort: 15008,
+			}
+		}
+	}
+	return w
+}
+
+func CreateFakeWorkload(params ...interface{}) *workloadapi.Workload {
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/common/fake_helpers.go:92]: I donot think it worth such complexity.

How about just passing some common fields, and also provide some customization func

```
func CreateFakeWorkload(a, b, c, fnc(*workloadapi.Workload) ) *workloadapi.Workload {
```

**Review by Insomniac2904 — COMMENTED** [review]
@hzxuzhonghu  can you check the changes that I did. I have used opt functions like you said.
Also can you tell me if we can do a lint-check locally, I seem to have most errors due to that.

**Comment by hzxuzhonghu** [comment]
we use golangci-lint to check the potential issues
```
golangci-lint run --out-format=github-actions --config=common/config/.golangci.yaml --out-format colored-line-number --verbose
```

The error is related to go imports, now we requires all `kmesh.net` pkg grouped together

File: pkg/controller/workload/common/fake_helpers.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+	"log"
+
+	"net/netip"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+	"kmesh.net/kmesh/api/v2/workloadapi"
+)
+
+func CreateFakeService(name, ip, waypoint string, optional ...interface{}) *workloadapi.Service {
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/common/fake_helpers.go:30]: I think this is not right, optional means you allow to pass in any types of values but the inner implement you only handle `*workloadapi.LoadBalancing`

File: pkg/controller/workload/common/fake_helpers.go [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import (
+	"fmt"
+	"log"
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/common/fake_helpers.go:21]: use kmesh log pkg instead

Comment (Insomniac2904) [reply pkg/controller/workload/common/fake_helpers.go:21]: I am using `logger := logger.NewLoggerScope("workload")` like this:
```
logger := logger.NewLoggerScope("workload")
	for _, opt := range opts {
		err := opt(workload)
		if err != nil {
			logger.Errorf("Error applying option: %v", err)
		}
	}
```

File: pkg/controller/workload/common/fake_helpers.go (hunk 6 above)
Comment (hzxuzhonghu) [reply pkg/controller/workload/common/fake_helpers.go:30]: Have you ppushed your new commit?

Comment (Insomniac2904) [reply pkg/controller/workload/common/fake_helpers.go]: give it a check now @hzxuzhonghu

File: pkg/controller/workload/common/fake_helpers.go [hunk 8]
```diff
@@ -42,28 +44,27 @@ func CreateFakeService(name, ip, waypoint string, optional ...interface{}) *work
...
+			{
+				ServicePort: 81,
+				TargetPort:  8180,
+			},
+			{
+				ServicePort: 82,
+				TargetPort:  82,
+			},
 		}
+		service.LoadBalancing = lbPolicy[0]
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/common/fake_helpers.go:62]: since here only one policy is needed, why use variable param?

Comment (Insomniac2904) [reply pkg/controller/workload/common/fake_helpers.go:62]: In waypoint_cache_test.go the func is called like this: `svc1 := common.CreateFakeService("svc1", "10.240.10.1", "")` so I used vardiac func, If I am make it fixed then I will have to pass `nil` during func call. will that be okay?

**Comment by hzxuzhonghu** [comment]
@Insomniac2904 can you fix the lint error, it is clear

**Comment by Insomniac2904** [comment]
> @Insomniac2904 can you fix the lint error, it is clear

Done

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**